### PR TITLE
Ap 167 3

### PR DESCRIPTION
**Pull request by ptravers**

### CODE REVIEW

File: src/bf/api/client/mod.rs [hunk 1]
```diff
@@ -720,7 +720,12 @@ impl Blackfynn {
             .into_iter()
             .enumerate()
             .map(|(id, file)| {
-                model::S3File::new(path.as_ref(), file.as_ref(), Some(Into::into(id as u64)))
+                let id = Some(Into::into(id as u64));
+
+                match is_directory_upload {
```
Comment (mikeswoods) [inline src/bf/api/client/mod.rs]: Probably just use an `if` here

Comment (ptravers) [reply src/bf/api/client/mod.rs]: I tried and it didn't like it not totally sure why it was some kind of type inferencing issue

Comment (mikeswoods) [reply src/bf/api/client/mod.rs]: Yeah, I'm curious about the error as well

Comment (ptravers) [reply src/bf/api/client/mod.rs]: ah I can add it back and find out give me a few minutes.

Comment (ptravers) [reply src/bf/api/client/mod.rs]: Hand to God it was failing. But now it's working...

Comment (ptravers) [reply src/bf/api/client/mod.rs]: @mattusifer @mikeswoods refactored to fix that now

File: src/bf/api/client/mod.rs [hunk 2]
```diff
@@ -720,7 +719,13 @@ impl Blackfynn {
             .into_iter()
             .enumerate()
             .map(|(id, file)| {
-                model::S3File::new(path.as_ref(), file.as_ref(), Some(Into::into(id as u64)))
+                let id = Some(Into::into(id as u64));
```
Comment (mikeswoods) [inline src/bf/api/client/mod.rs]: As just a matter of style (and for my own sanity), I think it is useful to provide the ID type when it is not immediately apparent what the type is:

`let id: Option<UploadId> = Some((id as u64).into());`
or 
`let id: Option<UploadId> = Some(Into::<UploadId>::into(id as u64));`

One cool facet of Rust is you can take a function that accepts a `self` argument, and explicitly pass it as the receiver: e.g. `x.into()` becomes `Into::into(x)`. This is necessary when you want to provide an explicit type for the hole in the invocation (`Into::<Y>::into(x)`), assuming the type accepts a type parameter.

**Pull request merged**